### PR TITLE
test_networking.test_vip test is flaky [2]

### PR DESCRIPTION
**Pull request by urbanserj**
## High Level Description

test_networking.test_vip test is flaky.

* Add logging
* Fix checking of deployment status
* Set adequate timeout

## Related Issues

  - [DCOS_OSS-1463](https://jira.dcos.io/browse/DCOS_OSS-1463) test_networking.test_vip test is flaky

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [x] Test Results: [link to CI job test results for component]
  - [x] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by urbanserj** [comment]
@mesosphere-mergebot bump-ee

**Comment by urbanserj** [comment]
@mesosphere-mergebot bump-ee

File: packages/dcos-integration-test/extra/test_networking.py [hunk 1]
```diff
@@ -68,22 +68,25 @@ def vip_app(container: marathon.Container, network: marathon.Network, host: str,
...
-    permutations = []
-    for container in [marathon.Container.NONE, marathon.Container.MESOS, marathon.Container.DOCKER]:
-        for vip_net in network_options:
-            for proxy_net in network_options:
-                if container != marathon.Container.DOCKER and marathon.Network.BRIDGE in (vip_net, proxy_net):
-                    # only DOCKER containers support BRIDGE network
-                    continue
-                permutations.append((container, vip_net, proxy_net))
-    return permutations
+    return [(container, vip_net, proxy_net)
```
Comment (gpaul) [inline packages/dcos-integration-test/extra/test_networking.py]: Lovely use of generators :+1:

File: packages/dcos-integration-test/extra/test_networking.py [hunk 2]
```diff
@@ -96,9 +99,11 @@ def test_vip(dcos_api_session, container: marathon.Container, vip_net: marathon.
     proxy container that will ping the origin container VIP and then assert
     that the expected origin app UUID was returned
     '''
-    failure_stack = []
-    for test in setup_vip_workload_tests(dcos_api_session, container, vip_net, proxy_net):
-        cmd, origin_app, proxy_app, proxy_net = test
+    errors = 0
+    tests = setup_vip_workload_tests(dcos_api_session, container, vip_net, proxy_net)
+    for vip, hosts, cmd, origin_app, proxy_app in tests:
+        log.info("Testing :: VIP: {}, Hosts: {}".format(vip, hosts))
```
Comment (gpaul) [inline packages/dcos-integration-test/extra/test_networking.py]: I was wishing for info logs here while trying to figure out which mesos task maps to the failed test. This is great.

**Comment by urbanserj** [comment]
I've run integration tests several times. Here are the results:

integration tests

**Comment by urbanserj** [comment]
@mellenburg @GoelDeepak Could you please review this PR?

**Comment by urbanserj** [comment]
@mesosphere-mergebot label [Ready For Review]

**Comment by asridharan** [comment]
@mellenburg @GoelDeepak could we prioritize this? Looks like the test are holding up. Seems like a genuine bug is getting fixed. This would definitely help us make a lot more progress.

**Comment by GoelDeepak** [comment]
lgtm

File: packages/dcos-integration-test/extra/test_networking.py (hunk 1 above)
Comment (jgehrcke) [reply packages/dcos-integration-test/extra/test_networking.py]: list comprehensions, you mean? :)

**Comment by jgehrcke** [comment]
Thank you, Sergey!

I have only peeked into the code, and I wonder if maybe next time https://docs.python.org/3.6/library/itertools.html could be helpful with `permutations()` and `combinations()`.

File: packages/dcos-integration-test/extra/test_networking.py (hunk 1 above)
Comment (gpaul) [reply packages/dcos-integration-test/extra/test_networking.py]: :D

**Comment by orsenthil** [comment]
Great work here @urbanserj . :) Thank you.

**Pull request merged**